lib/Jsrt/JsrtDiag.cpp

```diff
@@ -0,0 +1,688 @@
+                AssertMsg(false, "Failed to start debugging");
+                return JsErrorFatal; // Inconsistent state, we can't continue from here?
+            }
+
+            Js::ProbeContainer* probeContainer = debugContext->GetProbeContainer();
+            probeContainer->InitializeInlineBreakEngine(jsrtDebugManager);
+            probeContainer->InitializeDebuggerScriptOptionCallback(jsrtDebugManager);
+        }
+
+        return JsNoError;
+    });
+}
+
+CHAKRA_API JsDiagStopDebugging(
+    _In_ JsRuntimeHandle runtimeHandle,
+    _Out_ void** callbackState)
+{
+    return GlobalAPIWrapper([&]() -> JsErrorCode {
+
+        VALIDATE_INCOMING_RUNTIME_HANDLE(runtimeHandle);
+
+        PARAM_NOT_NULL(callbackState);
+
+        *callbackState = nullptr;
+
+        JsrtRuntime * runtime = JsrtRuntime::FromHandle(runtimeHandle);
+        ThreadContext * threadContext = runtime->GetThreadContext();
+
+        if (threadContext->GetRecycler() && threadContext->GetRecycler()->IsHeapEnumInProgress())
+        {
+            return JsErrorHeapEnumInProgress;
+        }
+        else if (threadContext->IsInThreadServiceCallback())
+        {
+            return JsErrorInThreadServiceCallback;
+        }
+        else if (threadContext->IsInScript())
+        {
+            return JsErrorRuntimeInUse;
+        }
+
+        ThreadContextScope scope(threadContext);
+
+        if (!scope.IsValid())
+        {
+            return JsErrorWrongThread;
+        }
+
+        JsrtDebugManager* jsrtDebugManager = runtime->GetJsrtDebugManager();
+
+        VALIDATE_IS_DEBUGGING(jsrtDebugManager);
+
+        for (Js::ScriptContext *scriptContext = runtime->GetThreadContext()->GetScriptContextList();
+        scriptContext != nullptr && !scriptContext->IsClosed();
+            scriptContext = scriptContext->next)
+        {
+            Assert(scriptContext->IsScriptContextInDebugMode());
+
+            if (FAILED(scriptContext->OnDebuggerDetached()))
+            {
+                AssertMsg(false, "Failed to stop debugging");
+                return JsErrorFatal; // Inconsistent state, we can't continue from here?
+            }
+
+            Js::DebugContext* debugContext = scriptContext->GetDebugContext();
+
+            Js::ProbeContainer* probeContainer = debugContext->GetProbeContainer();
+            probeContainer->UninstallInlineBreakpointProbe(nullptr);
+            probeContainer->UninstallDebuggerScriptOptionCallback();
+
+            jsrtDebugManager->ClearBreakpointDebugDocumentDictionary();
+        }
+
+        *callbackState = jsrtDebugManager->GetAndClearCallbackState();
+
+        return JsNoError;
+    });
+}
+
+CHAKRA_API JsDiagGetScripts(
+    _Out_ JsValueRef *scriptsArray)
+{
+    return ContextAPIWrapper<false>([&](Js::ScriptContext *scriptContext) -> JsErrorCode {
+
+        PARAM_NOT_NULL(scriptsArray);
+
+        *scriptsArray = JS_INVALID_REFERENCE;
+
+        JsrtContext *currentContext = JsrtContext::GetCurrent();
```

Comment:
We give scripts from all contexts see JsrtDebugManager::GetScripts. currentContetx is used to create the return JSON Array